mpy-cross/main.c

```diff
@@ -77,17 +90,22 @@ STATIC int compile_and_save(const char *file, const char *output_file, const cha
         cm.context = m_new_obj(mp_module_context_t);
         mp_compile_to_raw_code(&parse_tree, source_name, false, &cm);
 
-        vstr_t vstr;
-        vstr_init(&vstr, 16);
-        if (output_file == NULL) {
-            vstr_add_str(&vstr, file);
-            vstr_cut_tail_bytes(&vstr, 2);
-            vstr_add_str(&vstr, "mpy");
+        if (output_file != NULL && strcmp(output_file, "-") == 0) {
+            mp_raw_code_save(&cm, (mp_print_t *)&mp_stdout_print);
         } else {
-            vstr_add_str(&vstr, output_file);
+            vstr_t vstr;
+            vstr_init(&vstr, 16);
+            if (output_file == NULL) {
+                vstr_add_str(&vstr, file);
+                vstr_cut_tail_bytes(&vstr, 2);
+                vstr_add_str(&vstr, "mpy");
```

Comment:
Won't this make the output file `-.mpy` if the input file is `-`?

It might be better to make the default to output to stdout if the input is stdin and no output argument is given.

Reply:
Indeed, I've changed it accordingly.